### DISCUSSION TITLE
Vídeo 02 - Configuração - Pergunta adicional

### DISCUSSION THREAD

**Discussion by thpoiani** [thpoiani/lab-quarkus#3] [Q&A] [answered]
Seguindo o padrão do projeto, quais alterações são necessárias em **cicd-build.sh** e **docker-compose.yml** para que seja executado builds para **JVM** ou **Native Image**?

**Comment by didifive** [comment, marked as answer]
O script em **cicd-build.sh** irá executar o Docker Compose Build conforme o que está configurado em **docker-compose.yml** na linha 17:
```shell
TAG=$APP_VERSION docker compose build --no-cache "$APP"
```

Para mudar opção de build, é necessário ir ao arquivo **docker-compose.yml**, procurar a configuração de  docker compose referente à aplicação e alterar `dockerfile` aninhado à `build`, sendo:

- `dockerfile: src/main/docker/Dockerfile.native` apontando para Dockerfile do projeto para **Native Image**
- `dockerfile: src/main/docker/Dockerfile.jvm`  apontando para Dockerfile do projeto para **JVM**

Exemplo de configuração para docker compose da aplicação **election-management**  com build para **JVM**:
```docker
  election-management:
    extends:
      file: common.yml
      service: quarkus
    build:
      context: ./election-management
      dockerfile: src/main/docker/Dockerfile.jvm
    image: dio/election-management:${TAG:-1.0.0}
    labels:
      - traefik.http.routers.election-management.rule=Host(`vote.dio.localhost`) && PathPrefix(`/api`)
      - traefik.http.routers.election-management.entrypoints=web
```
